code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -0,0 +1,213 @@
+			qdel(src)
+	else
+		if(user)
+			user << "<span class='danger'>Looks like you didn't win anything this time...next time though, right?</span>"
+*/
+//Gluttony
+
+/obj/effect/gluttony
+	name = "gluttony's wall"
+	desc = "Only those who truly indulge may pass."
+	anchored = 1
+	density = 1
+	icon_state = "blob"
+	icon = 'icons/mob/blob.dmi'
+
+/obj/effect/gluttony/CanPass(atom/movable/mover, turf/target, height=0)//So bullets will fly over and stuff.
+	if(height==0)
+		return 1
+	if(istype(mover, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = mover
+		if(H.nutrition >= NUTRITION_LEVEL_FAT)
+			return 1
+		else
+			H << "<span class='danger'><B>You're not gluttonous enough to pass this barrier!</B></span>"
+	else
+		return 0
+
+//Pride
+
+/obj/structure/mirror/magic/pride
+	name = "pride's mirror"
+	desc = "Pride cometh before the..."
+	icon_state = "magic_mirror"
+/* TODO:curse and chasm
+/obj/structure/mirror/magic/pride/curse(mob/user)
+	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as their hand leaves the mirror!</B></span>")
+	var/turf/T = get_turf(user)
+	T.ChangeTurf(/turf/simulated/chasm/straight_down)
+	var/turf/simulated/chasm/straight_down/C = T
+	C.drop(user)*/
+
+//Sloth - I'll finish this item later
+
+//Envy
+
+/obj/item/weapon/knife/envy
+	name = "envy's knife"
+	desc = "Their success will be yours."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "render"
+	item_state = "render"
+	force = 18
+	throwforce = 10
+	w_class = 3
+	hitsound = 'sound/weapons/bladeslice.ogg'
+/*TODO: Attack
+/obj/item/weapon/knife/envy/afterattack(atom/movable/AM, mob/living/carbon/human/user, proximity)
+	..()
+	if(!proximity)
+		return
+	if(!istype(user))
+		return
+	if(istype(AM, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = AM
+		if(user.real_name != H.dna.real_name)
+			user.real_name = H.dna.real_name
+			H.dna.transfer_identity(user, transfer_SE=1)
+			user.updateappearance(mutcolor_update=1)
+			user.domutcheck()
+			user << "You assume the face of [H]. Are you satisfied?"*/
+
+///Ash Walkers
+
+/mob/living/simple_animal/hostile/spawner/ash_walker
+	name = "ash walker nest"
+	desc = "A nest built around a necropolis tendril. The eggs seem to grow unnaturally fast..."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "ash_walker_nest"
+	icon_living = "ash_walker_nest"
+	health = 200
+	maxHealth = 200
+	loot = list(/obj/effect/gibspawner, /obj/item/device/assembly/signaler/anomaly)
+	del_on_death = 1
+	var/meat_counter
+
+/mob/living/simple_animal/hostile/spawner/ash_walker/Life()
+	..()
+	if(!stat)
+		consume()
+		spawn_mob()
+
+/mob/living/simple_animal/hostile/spawner/ash_walker/proc/consume()
+	for(var/mob/living/H in view(src,1)) //Only for corpse right next to/on same tile
+		if(H.stat)
+			visible_message("<span class='warning'>Tendrils reach out from \the [src.name] pulling [H] in! Blood seeps over the eggs as [H] is devoured.</span>")
```

Comment:
`Devoured` is one of the fun examples where really Americans if they were consistent, they'd spell it `devored`. ;)